spec/models/user_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'Validations For the user model' do
+    before(:each) do
+      subject { User.new(name: 'Joseph', posts_counter: 3) }
+    end
+
+    before { subject.save }
+
+    it 'name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be present' do
+      subject.posts_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should allow valid values' do
+      subject.posts_counter = 34
+      expect(subject).to_not be_valid
+    end
+  end
+end
```

Comment:
- [ ] Please make sure to test all methods implemented in the `User` model, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods) requirement.